packages/video_player_avplay/lib/video_player.dart

```diff
@@ -1246,26 +1246,29 @@ class _VideoPlayerState extends State<VideoPlayer> {
     WidgetsBinding.instance.addPostFrameCallback(_afterFrameLayout);
   }
 
+  bool _isInvalid(double value) {
+    return value.isInfinite || value.isNaN;
+  }
+
   void _afterFrameLayout(_) {
     if (widget.controller.value.isInitialized) {
-      final Rect currentRect = _getCurrentRect();
-      if (currentRect != Rect.zero && _playerRect != currentRect) {
+      final Rect rect = _getCurrentRect();
+      if (rect != Rect.zero && _playerRect != rect) {
+        final double offsetLeft = rect.left - rect.left.floor();
+        final double offsetTop = rect.top - rect.top.floor();
+        final double offsetWidth = rect.width.ceil() - rect.width;
+        final double offsetHeight = rect.height.ceil() - rect.height;
+        final int left = _isInvalid(rect.left) ? 0 : rect.left.floor();
+        final int top = _isInvalid(rect.top) ? 0 : rect.top.floor();
+        final int width = _isInvalid(rect.width)
+            ? 1
+            : rect.width.ceil() + ((offsetLeft > offsetWidth) ? 1 : 0);
+        final int height = _isInvalid(rect.height)
+            ? 1
+            : rect.height.ceil() + ((offsetTop > offsetHeight) ? 1 : 0);
```

Comment:
Is this logic the same as PlusPlayer's surface logic ?
In some cases, it might be 1 pixel larger.
umm.. wouldn't it be more appropriate to use just `round() +- 1`? 


```dart
void main() {
  double left =   9.999999999;   // 10
  double width = 20.00000001;  // 20

  final double offsetLeft =
      left - left.floor(); // 9.9999 - 9 = 0.9999...
  final double offsetWidth =
      width.ceil() - width; // 21 - 20.00001 = 0.9999...

  var final_left = left.floor(); // 9
  var final_width = width.ceil() + ((offsetLeft > offsetWidth) ? 1 : 0); // 21 + 0 or 1
  
  print(left);
  print(width);
  
  print(final_left);
  print(final_width);
}
```